scripts/ai_api_helper.py

```
@@ -2,6 +2,66 @@
+        return False, f"Syntax error: {str(e)}"
+    except Exception as e:
+        return False, f"Validation error: {str(e)}"
+
+def exponential_backoff_retry(func, max_retries=3):
+    """Retry function with exponential backoff"""
+    for attempt in range(max_retries):
+        try:
+            return func()
+        except Exception as e:
+            if attempt == max_retries - 1:
+                raise e
```

Ruff (0.12.2)

Use `raise` without specifying exception name

Remove exception name

(TRY201)